docs/CADL.md

```diff
@@ -0,0 +1,93 @@
+# Design of CADL Support
+
+## Basic Terms
+- CADL pipeline: the process and components of C# code generation from cloud services API specification written in [CADL](https://github.com/microsoft/cadl).
+- Swagger pipeline: the process and components of C# code generation from REST API specification written in [OpenAPI](https://swagger.io/).
+- generator: autorest.csharp per se, last part of both pipelines that is directly responsible for code generation.
+- generator input: file or stream with input for autorest.csharp, produced by previous part in the pipeline.
+- CADL C# Emitter: part of the CADL pipeline. Emitters are used by CADL compiler to output result of its work.
+- CodeModel.yaml: generator input in Swagger pipeline.
+- CodeModel: deserialized object representation of CodeModel.yaml in autorest.csharp.
+- cadl.json: generator input in CADL pipeline produced by CADL emitter.
+- InputModel: deserialized object representation of cadl.json in autorest.csharp.
```

Comment:
Here, the name inputModel & outputModel conflict with the model name used in ModelGeneration ( Input/Output/RoundTrip models).  is there any plan to do refactor/rename to avoid confusion?

Reply:
Yeah, this name conflict exists for years. We have model layers in generator itself and we generate models. I'm open to suggestions.

Reply:
I created https://github.com/Azure/autorest.csharp/issues/2712 and we could address that in MQ.